### PR TITLE
[RPD-132] Move `matcha provision` existing deployment validations to the start of the command

### PR DESCRIPTION
**Pull request by d-lowl**
Ask for provisioning variables only after checking if there's an existing configuration.

Do not ask for the variables, if the configuration exists, and the user tells to not override it. Allow for empty variables in validation, so we do not make the CLI options mandatory. Prompt and fill in the variables only if they were not provided as CLI options.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [x] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/cli/_validation.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ def region_typer_callback(region: str) -> str:
     Returns:
         str: the region after checks are passed.
     """
+    if not region:
```
Comment (osw282) [inline src/matcha_ml/cli/_validation.py:142]: Do you prefer this or:

```
if region:

    try:
        region_validation(region)
    except MatchaInputError as e:
        raise BadParameter(str(e))

return region
```

**Review by osw282 — APPROVED** [review]
LGTM, just one suggestion! 🚀

File: src/matcha_ml/cli/_validation.py (hunk 1 above)
Comment (d-lowl) [reply src/matcha_ml/cli/_validation.py:142]: `not region` check is a bit more specific in my opinion (we specifically check for the empty string, as we know that it maybe empty when no CLI option is passed -- so we do not check the rest --, and it cannot be empty after the prompt)

**Comment by d-lowl** [comment]
@Christopher-Norman the changes do not change what test cases are required (both CLI options and in-app prompts are already covered)

File: src/matcha_ml/cli/_validation.py (hunk 1 above)
Comment (osw282) [reply src/matcha_ml/cli/_validation.py:142]: Sounds good to me.

**Review by Christopher-Norman — APPROVED** [review]
LGTM

**Review by JonoCX — APPROVED** [review]
LGTM - nice 🦾

**Pull request merged**